pkg/process/filter.go

```diff
@@ -2,28 +2,57 @@ package process
 
 import (
 	"fmt"
+	"k8s.io/apimachinery/pkg/labels"
 	"regexp"
 	"strings"
 
 	"github.com/grafana/tanka/pkg/kubernetes/manifest"
 )
 
-// Filter returns all elements of the list that match at least one expression
+type FilterOptions struct {
+	Exprs    Matchers
+	Selector labels.Selector
+}
+
+type FilterOption func(options *FilterOptions)
+
+// FilterWithOptions returns all elements of the list that match at least one expression
 // and are not ignored
-func Filter(list manifest.List, exprs Matchers) manifest.List {
+func FilterWithOptions(list manifest.List, options ...FilterOption) manifest.List {
```

Comment:
Do you have a specific reason to treat Matchers and the Selector differently?

The idea for Matchers was to account for any kind of filtering that needs to happen, though arguably the interface isn't quite there as it filters on the kind/name string.

I think it would be a good opportunity to change that, maybe like so:

```go
type Matcher interface {
	Match(manifest.Manifest) bool
}
```

That way your selector filter could be a matcher injected by the main package.

Wdyt?

Reply:
How do you feel about using the code as it stands, and performing a larger refactor as a separate PR?

Reply:
Hey!

If I understand your suggestion correctly, that would replace current `-t` with a new system, turning `-t pod/foo` into `-l kind==pod metadata.name==foo`?
I'd kinda like to avoid such a breaking change.

Also having to implement custom lexer/parser systems appears a bit overkill for what we are trying to achieve.

> How do you feel about using the code as it stands, and performing a larger refactor as a separate PR?
Sure, my only concern is that you had to deprecate certain parts of the public facing API, which I think we could avoid.

I'd suggest taking the middle route here and implementing both regex and label matching using the interface I suggested above. That should result in fairly minimal changes.

Wdyt?